mitodl/ocw-studio
# don't update fields that don't exist in resource data

#### Pre-Flight checklist


#### What are the relevant tickets?
none

#### What's this PR do?
This pr updates the `overwrite_ocw_course_content` command to not create any fields that don't exist in the data in the content bucket we are importing from. This is needed because we want to add start_time and end_time to the metadata for video resources but not any other type of content.

#### How should this be manually tested?
Run 
`docker-compose run web ./manage.py import_ocw_course_sites --bucket ocw-to-hugo-output-qa --filter 18-01sc-single-variable-calculus-fall-2010`

Go to a video resource in 18-01sc-single-variable-calculus-fall-2010 and change the youtube id. 

Run
`docker-compose run web ./manage.py overwrite_ocw_course_content --bucket ocw-to-hugo-output-qa --filter 18-01sc-single-variable-calculus-fall-2010 --content-field metadata.video_metadata.youtube_id` and verify that the youtube id is changed back to it's original value

Run
`docker-compose run web ./manage.py overwrite_ocw_course_content --bucket ocw-to-hugo-output-qa --filter 18-01sc-single-variable-calculus-fall-2010 --content-field metadata.bad_key`

Go to the admin console and search for resources for 18-01sc-single-variable-calculus-fall-2010. Verify that they don't have  `bad_key` set in the metadata.
